src/appleseed/renderer/modeling/object/meshobject.cpp

```diff
@@ -144,6 +145,53 @@ Lazy<RegionKit>& MeshObject::get_region_kit()
     return impl->m_lazy_region_kit;
 }
 
+void MeshObject::rasterize(ObjectRasterizer& rasterizer) const
+{
+    rasterizer.begin_object();
+
+    for (const auto& prim : impl->m_tess.m_primitives)
+    {
+        const auto& v0 = impl->m_tess.m_vertices[prim.m_v0];
+        const auto& v1 = impl->m_tess.m_vertices[prim.m_v1];
+        const auto& v2 = impl->m_tess.m_vertices[prim.m_v2];
+
+        // todo: check that vertex normals are available.
+        const auto& n0 = impl->m_tess.m_vertex_normals[prim.m_n0];
+        const auto& n1 = impl->m_tess.m_vertex_normals[prim.m_n1];
+        const auto& n2 = impl->m_tess.m_vertex_normals[prim.m_n2];
+
+        ObjectRasterizer::Triangle triangle;
+
+        triangle.m_v0[0] = v0[0];
```

Comment:
Please, move this into triangle constructor (e.g. triangle(v0, v1, v2, n0, n1, n2))
Or to another "factory" function (e.g. init_triangle(triangle, vo, vq, v2, n0, n1, n2) or even triangle = construct_triangle(v0, v1, v2, n0, n1, n2))